AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/fragments/moviesList/FragmentMoviesList.kt

```diff
@@ -1,36 +1,61 @@
 package com.example.androidhomeapplication.fragments.moviesList
 
 import android.os.Bundle
+import android.util.Log
 import android.view.View
+import android.widget.Toast
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentFactory
 import androidx.recyclerview.widget.GridLayoutManager
 import by.kirich1409.viewbindingdelegate.viewBinding
-import com.example.androidhomeapplication.DataGenerator
+import com.android.academy.fundamentals.homework.data.MovieRepository
+import com.android.academy.fundamentals.homework.data.MovieRepositoryProvider
 import com.example.androidhomeapplication.R
 import com.example.androidhomeapplication.databinding.FragmentMoviesListBinding
 import com.example.androidhomeapplication.fragments.movieDetails.MovieDetailsScreen
-import com.example.androidhomeapplication.models.MovieData
 import com.example.androidhomeapplication.navigation.RouterProvider
 import com.github.terrakok.cicerone.androidx.FragmentScreen
+import kotlinx.coroutines.*
 
 class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
     private val binding by viewBinding(FragmentMoviesListBinding::bind)
+    private var scope: CoroutineScope? = null
+    private val movieRepository: MovieRepository get() = (activity?.application as MovieRepositoryProvider).movieRepository
```

Comment:
what about extension for Fragment:

private val movieRepository: MovieRepository get() = (activity?.application as MovieRepositoryProvider).movieRepository

and reuse it